libasherah_test.go

```diff
@@ -0,0 +1,730 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/godaddy/cobhan-go"
+)
+
+func validSetupForTesting(t *testing.T) {
+	config := AsherahConfig{}
+
+	config.KmsType = "static"
+	config.ServiceName = "TestService"
+	config.ProductID = "TestProduct"
+	config.Metastore = "memory"
+	config.SessionCache = true
+	config.SessionCacheDuration = 1000
+	config.SessionCacheMaxSize = 2
+	config.ExpireAfter = 1000
+	config.CheckInterval = 1000
+	config.DebugOutput = true
+	config.RegionMapStr = "region1=arn1,region2=arn2"
+
+	jsonBytes, err := json.Marshal(config)
+	if err != nil {
+		t.Error(err)
+	}
+
+	buf := cobhan.AllocateBuffer(1024)
+
+	result := cobhan.BytesToBufferSafe(jsonBytes, &buf)
+	if result != ERR_NONE {
+		t.Error(fmt.Sprintf("BytesToBufferSafe returned %v", result))
+	}
+
+	SetupJson(cobhan.Ptr(&buf))
+}
+
+func testAllocateStringBuffer(t *testing.T, str string) []byte {
```

Comment:
Is it a convention for tests to use downcase `test` prefix for helper functions that have assertions? Otherwise, to avoid confusion, what about using just `allocateStringBuffer`? If it is a convention, we should probably rename `validSetupForTesting` function to `testSetupJson`?

Reply:
This is a local function so it uses downcase.  The Test functions are uppercase.  The prefix test means something in Go.